### PR TITLE
Dev #7367: 'Other answer' at a certain position in the list

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/views/survey/questions/answer/list_dropdown/config.xml [hunk 1]
```diff
@@ -142,6 +142,49 @@
             <readonly></readonly>
             <readonly_when_active></readonly_when_active>
         </attribute>
+        <attribute>
+            <name>other_position</name>
+            <category>Display</category>
+            <sortorder>150</sortorder>
+            <inputtype>singleselect</inputtype>
```
Comment (Shnoulle) [inline application/views/survey/questions/answer/list_dropdown/config.xml:149]: Why use integer ? Why not simple string ?

```
           <options>
                <option>
                    <value>begin</value>
                    <text>At beginning</text>
                </option>
                <option>
                    <value>default</value>
                    <text>Before No Answer</text>
                </option>
                <option>
                    <value>end</value>
                    <text>At end</text>
                </option>
                <option>
                    <value>specific</value>
                    <text>After specific answer option</text>
                </option>
           </options>
           <default>default</default>
```

File: application/core/QuestionTypes/ListDropdown/RenderListDropdown.php [hunk 2]
```diff
@@ -21,9 +21,26 @@ class RenderListDropdown extends QuestionBaseRenderer
...
+
+    /** @var string the code of the answer after which the 'Other' option should be placed (if $otherPosition == 3) */
+    protected $answerBeforeOther;
+
+    /** @var boolean indicates if 'Other' option has already been rendered */
+    protected $otherRendered = false;
     
     private $iRowNum = 0;
 
+    const OTHER_POS_BEFORE_NOANSWER = 0;
```
Comment (Shnoulle) [inline application/core/QuestionTypes/ListDropdown/RenderListDropdown.php:39]: `const OTHER_POS_DEFAULT = 'default';` etc …

**Review by Shnoulle — CHANGES_REQUESTED** [review]
For discussion : 

1. I prefer string than integer 
2. No test survey ? New functionality : new test. And here a sample lss with the 3 question type + some option set.

**Comment by gabrieljenik** [comment]
Test added

**Comment by Shnoulle** [comment]
Thank you for string in place of integer .
You're OK too ?

**Review by Shnoulle — APPROVED** [review]

File: application/core/QuestionTypes/ListDropdown/RenderListDropdown.php [hunk 3]
```diff
@@ -21,9 +21,26 @@ class RenderListDropdown extends QuestionBaseRenderer
...
+
+    /** @var string the code of the answer after which the 'Other' option should be placed (if $otherPosition == 3) */
+    protected $answerBeforeOther;
+
+    /** @var boolean indicates if 'Other' option has already been rendered */
+    protected $otherRendered = false;
     
     private $iRowNum = 0;
 
+    const OTHER_POS_BEFORE_NOANSWER = 'default';
```
Comment (olleharstedt) [inline application/core/QuestionTypes/ListDropdown/RenderListDropdown.php:39]: These and other lines seems duplicated in lots of classes? Can we move it out to a trait instead?

Comment (gabrieljenik) [reply application/core/QuestionTypes/ListDropdown/RenderListDropdown.php:39]: I will create a new DEV ticket for that

Comment (gabrieljenik) [reply application/core/QuestionTypes/ListDropdown/RenderListDropdown.php:39]: https://bugs.limesurvey.org/view.php?id=18489

**Pull request merged**